packages/sftp-client/tests/test-index.js

```diff
@@ -0,0 +1,214 @@
+  );
+
+  await S3.putFile(
+    process.env.system_bucket,
+    `${process.env.stackName}/crypto/private.pem`,
+    require.resolve('@cumulus/test-data/keys/s3_key_pair_provider_private.pem')
+  );
+
+  await S3.putFile(
+    process.env.system_bucket,
+    `${process.env.stackName}/crypto/public.pub`,
+    require.resolve('@cumulus/test-data/keys/s3_key_pair_provider_public.pub')
+  );
+
+  const createKeyResponse = await KMS.createKey();
+  t.context.kmsKeyId = createKeyResponse.KeyMetadata.KeyId;
+});
+
+test.after.always(() => S3.recursivelyDeleteS3Bucket(process.env.system_bucket));
+
+test('SftpClient supports plaintext usernames and passwords', async (t) => {
+  const sftpClient = new SftpClient({
+    host: '127.0.0.1',
+    port: '2222',
+    username: 'user',
+    password: 'password',
+    encrypted: false
+  });
+
+  const files = await sftpClient.list('/');
+  const fileNames = files.map((f) => f.name);
+
+  t.true(fileNames.includes('index.html'));
+});
+
+test('SftpClient supports S3-keypair-encrypted usernames and passwords', async (t) => {
+  const username = await S3KeyPairProvider.encrypt('user');
+  const password = await S3KeyPairProvider.encrypt('password');
+
+  const sftpClient = new SftpClient({
+    host: '127.0.0.1',
+    port: '2222',
+    username,
+    password,
+    encrypted: true
+  });
+
+  const files = await sftpClient.list('/');
+  const fileNames = files.map((f) => f.name);
+
+  t.true(fileNames.includes('index.html'));
+});
+
+test('SftpClient supports KMS-encrypted usernames and passwords', async (t) => {
+  const username = await KMS.encrypt(t.context.kmsKeyId, 'user');
+  const password = await KMS.encrypt(t.context.kmsKeyId, 'password');
+
+  const sftpClient = new SftpClient({
+    host: '127.0.0.1',
+    port: '2222',
+    username,
+    password,
+    encrypted: true
+  });
+
+  const files = await sftpClient.list('/');
+  const fileNames = files.map((f) => f.name);
+
+  t.true(fileNames.includes('index.html'));
+});
+
+test('SftpClient supports unencrypted private keys', async (t) => {
+  const sftpClient = new SftpClient({
+    host: '127.0.0.1',
+    port: '2222',
+    username: 'user',
+    privateKey: 'ssh_client_rsa_key'
+  });
+
+  const files = await sftpClient.list('/');
+  const fileNames = files.map((f) => f.name);
+
+  t.true(fileNames.includes('index.html'));
+});
+
+test('SftpClient supports KMS-encrypted private keys', async (t) => {
+  const unencryptedPrivateKey = await readFile(
+    require.resolve('@cumulus/test-data/keys/ssh_client_rsa_key'),
+    'utf8'
+  );
+
+  const privateKeyName = randomString();
+
+  await S3.s3PutObject({
+    Bucket: process.env.system_bucket,
+    Key: `${process.env.stackName}/crypto/${privateKeyName}`,
+    Body: await KMS.encrypt(t.context.kmsKeyId, unencryptedPrivateKey)
+  });
+
+  const sftpClient = new SftpClient({
+    host: '127.0.0.1',
+    port: '2222',
+    username: 'user',
+    cmKeyId: 'yes',
```

Comment:
I guess any truthy value will work, but for the sake of a test as documentation, should this be `true?

Reply:
According to the provider schema, it should be a string.

```js
cmKeyId: {
  type: 'string',
  description: 'AWS KMS Customer Master Key arn or alias'
}
```

That being said, the value of `cmKeyId` is never actually used. There's definitely some refactoring there that should be done but, since I didn't add that code as part of this ticket, I thought that it was out of scope.

Reply:
I see. Because `cmKeyId` actually will come from a provider object when the `SftpClient` is initialized?

https://github.com/nasa/cumulus/pull/1446/files?file-filters%5B%5D=.js&file-filters%5B%5D=.tf&file-filters%5B%5D=.yml#diff-bc5d232e73466353582fee67f9820ca8R21

Reply:
Correct.